### PR TITLE
[NTGDI][FONT] Don't rely FT_FaceRec.style_name

### PR DESCRIPTION
**Pull request by katahiromz**
## Purpose

JIRA issue: [CORE-9614](https://jira.reactos.org/browse/CORE-9614)

- Delete `ItalicFromStyle` and `WeightFromStyle` functions.
- Don't use `FT_FaceRec.style_name` for font style but `pOS2->fsSelection`, `pOS2->usWeightClass`, `WinFNT.italic` and `WinFNT.weight`.
- Don't use the bitmap glyph as possible if the glyph will be oblique'd.

### CODE REVIEW

**Comment by katahiromz** [comment]
it-seems-working
It seems working...

**Comment by katahiromz** [comment]
DumpMetrics2
I cannot understand why `OriginalItalic` is `true`.

File: win32ss/gdi/ntgdi/freetype.c [hunk 1]
```diff
@@ -1279,11 +1223,34 @@ IntGdiLoadFontsFromMemory(PGDI_LOAD_FONT pLoadFont)
...
-            FontGDI->OriginalItalic = ItalicFromStyle(Face->style_name);
+            FontGDI->OriginalItalic = FALSE;
             FontGDI->RequestItalic = FALSE;
-            FontGDI->OriginalWeight = WeightFromStyle(Face->style_name);
+            FontGDI->OriginalWeight = FW_NORMAL;
             FontGDI->RequestWeight = FW_NORMAL;
 
+            if (FT_IS_SFNT(Face))
+            {
+                IntLockFreeType();
```
Comment (learn-more) [inline win32ss/gdi/ntgdi/freetype.c]: I would move the locking to outside the `if (FT_IS_SFNT(Face))` block (so just lock this entire new block).
As for why Italic does not work, not sure :(

Comment (katahiromz) [reply win32ss/gdi/ntgdi/freetype.c]: Maybe `FT_GlyphSlot_Oblique` doesn't support bitmap glyphs. Should we manually transform the bitmap glyphs?

Comment (katahiromz) [reply win32ss/gdi/ntgdi/freetype.c]: We can use `FT_LOAD_NO_BITMAP` flag to load non-bitmap glyphs.

Comment (katahiromz) [reply win32ss/gdi/ntgdi/freetype.c]: Gotcha!
gotcha

**Comment by katahiromz** [comment]
Gotcha!
gotcha

**Comment by katahiromz** [comment]
Let's go to the testing stage!

**Comment by katahiromz** [comment]
The results:
https://reactos.org/testman/compare.php?ids=70071,70073

**Pull request merged**